File: PlacementCellBackend/Models/CompanyEmployee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlacementCellBackend.Models
{
    public class Companyemployee
    {
        [Key]
        public string employeeid { get; set; } = string.Empty;
        public string name { get; set; } = string.Empty;

        public string designation { get; set; } = string.Empty;

        public string email { get; set; } = string.Empty;

        [Required]
        public string companyid { get; set; } = string.Empty;

        [ForeignKey("companyid")]
        public Company? Company { get; set; }
    }
}
    

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/models/CompanyEmployee.java
package com.example.placementicsbackend.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "companyemployee")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CompanyEmployee {

    @Id
    @Column(name = "EmployeeId")
    private String employeeId;

    @Column(name = "name", nullable = false)
    private String name = "";

    @Column(name = "Designation", nullable = false)
    private String designation = "";

    @Column(name = "Email", nullable = false)
    private String email = "";

    @Column(name = "CompanyId", nullable = false)
    private String companyId;

    @Column(name = "ProfileUrl", nullable = false)
    private String profileUrl = "";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CompanyId", insertable = false, updatable = false)
    private Company company;
}
